hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,248 @@
+        else:
+            a_string = a_string.replace("[", "['").replace("]", "']")
+            a_string = a_string.replace(",", "','")
+            a_list = ast.literal_eval(a_string)
+            a_list = [ele.strip() for ele in a_list]
+            return a_list
+
+    def _string_to_dict(self, a_string):
+        """Converts a string representation of dictionary
+        into a dictionary type object.
+
+        Args:
+            a_string: dictionary recieved as type string
+                eg. Strings starting with '{' and ending with '}'
+        Returns: Object of dictionary type.
+        Sample String:
+        '{abc:123, def:xyz,    space   :    lot of it , 2345:some number }'
+        """
+        if a_string == '{}':
+            a_dict = ast.literal_eval(a_string)
+            return a_dict
+        else:
+            a_string = a_string.replace("{", "{'").replace("}", "'}")
+            a_string = a_string.replace(":", "':'").replace(",", "','")
+            a_dict = ast.literal_eval(a_string)
+            a_dict = {k.strip(): v.strip() for k, v in a_dict.iteritems()}
+            return a_dict
+
+    def _interface_info(self, port):
+        """Gets latest configuration of port from switch.
+
+        Args:
+            port: Valid port name
+        Returns: configuration status of the port
+        """
+        # This function is differnet then `ovs_connect` as it uses
+        # subprocess.check_output because it only needs read info from switch
+        # and pass the output to calling funtion.
+        shell_cmd = "sudo ovs-vsctl list port {port}".format(port=port)
+        args = shlex.split(shell_cmd)
+        try:
+            output = subprocess.check_output(args)
+        except subprocess.CalledProcessError as e:
+            logger.error(" %s ", e)
+            raise SwitchError('Ovs command failed: ')
+        output = output.split('\n')
+        output.remove('')
+        i_info = dict(s.split(':') for s in output)
+        i_info = {k.strip(): v.strip() for k, v in i_info.iteritems()}
+        # above statement removes extra white spaces from the keys and values.
+        # That is important since other calls will be querying this dictionary.
+        for x in i_info.keys():
+            if i_info[x][0] == "{":
+                i_info[x] = self._string_to_dict(i_info[x])
+            elif i_info[x][0] == "[":
+                i_info[x] = self._string_to_list(i_info[x])
+        return i_info
+
+    def _remove_native_vlan(self, port):
+        """Removes native vlan from a trunked port.
+        If it is the last vlan to be removed, it disables the port and
+        reverts its state to default configuration
+        Args:
+            port: Valid switch port
+        Returns: if successful None else error message
+        """
+        port_info = self._interface_info(port)
+        vlan_id = port_info['tag']
+        shell_cmd = "sudo ovs-vsctl remove port {port} tag {vlan_id}".format(
```

Comment:
I think it makes it for a readable code. Read as command line send as a list. 
Helps in case of debugging too. 
Eg. Past the command line as it is and then programmatically convert it to the list. I would like to keep it this way, if there is no strong reason against it.